cpp/dolfinx/la/SparsityPattern.cpp

```diff
@@ -238,83 +240,96 @@ void SparsityPattern::assemble()
   for (std::int64_t global_i : ghosts1)
     global_to_local.insert({global_i, local_i++});
 
-  // For each ghost row, pack and send (global row, global col) pairs to send to
-  // neighborhood
-  std::vector<std::int64_t> ghost_data;
+  // Get ghost->owner communicator for rows
+  MPI_Comm comm = _index_maps[0]->comm(common::IndexMap::Direction::reverse);
+  int indegree_rev(-1), outdegree_rev(-2), weighted_rev(-1);
+  MPI_Dist_graph_neighbors_count(comm, &indegree_rev, &outdegree_rev,
+                                 &weighted_rev);
+  const auto [src_ranks, dest_ranks] = dolfinx::MPI::neighbors(comm);
+
+  // Global-to-neigbourhood map for destination ranks
+  std::map<int, std::int32_t> dest_proc_to_neighbor;
```

Comment:
Maybe an `unordered_map` would be faster?
It may not matter here because the map is small (dest_ranks size).

Reply:
`std::map` is probably better here because it should be quite small.